src/app/modules/common/services.js

```diff
@@ -630,6 +631,24 @@ angular.module('BloomLibraryApp.services', ['restangular'])
                 return tag;
             }
         };
+
+        this.isSystemTag = function(tag) {
+            var regex = new RegExp('^system:');
+            return regex.test(tag);
+        };
+
+        this.hideSystemTags = function(book) {
+            if(book.tags) {
+                for (var i = 0; i < book.tags.length; i++) {
+                    var tag = book.tags[i];
+                    if (this.isSystemTag(tag)) {
+                        book.tags.splice(i, 1);
+                        //Since we spliced out a tag, re-check that index
+                        i--;
```

Comment:
A slightly easier way to avoid this problem is to make 'i' count down from book.tags.length-1 to 0. Probably not worth another commit though, unless other things are being changed.